Sources/Components/Feed/InAppFeedViewModel.swift

```diff
@@ -0,0 +1,368 @@
+            }
+        }
+        
+        public func markAsRead(_ item: Knock.FeedItem) async {
+            guard item.read_at == nil else { return }
+            do {
+                let _ = try await Knock.shared.messageModule.updateMessageStatus(messageId: item.id, status: .read)
+                
+                if let index = feed.entries.firstIndex(where: { $0.id == item.id }) {
+                    await MainActor.run {
+                        feed.entries[index].read_at = Date()
+                    }
+                }
+                
+                await fetchNewMetaData()
+            } catch {
+                Knock.shared.log(type: .error, category: .feed, message: "error in markAsRead: \(error.localizedDescription)")
+            }
+        }
+        
+        public func markAsUnread(_ item: Knock.FeedItem) async {
+            guard item.read_at != nil else { return }
+            do {
+                let _ = try await Knock.shared.messageModule.updateMessageStatus(messageId: item.id, status: .unread)
+                
+                if let index = feed.entries.firstIndex(where: { $0.id == item.id }) {
+                    await MainActor.run {
+                        feed.entries[index].read_at = nil
+                    }
+                }
+                
+                await fetchNewMetaData()
+            } catch {
+                Knock.shared.log(type: .error, category: .feed, message: "error in markAsUnread: \(error.localizedDescription)")
+            }
+        }
+        
+        public func markAsInteracted(_ item: Knock.FeedItem) async {
+            do {
+                let _ = try await Knock.shared.messageModule.updateMessageStatus(messageId: item.id, status: .interacted)
+            } catch {
+                Knock.shared.log(type: .error, category: .feed, message: "error in markAsInteracted: \(error.localizedDescription)")
+            }
+        }
+        
+        public func optimisticallyBulkUpdateStatus(status: Knock.KnockMessageStatusUpdateType) {
+            let date = Date()
+            var meta = feed.meta
+            DispatchQueue.main.async {
+                self.feed.entries = self.feed.entries.map { item in
+                    var newItem = item
+                    switch status {
+                    case .seen:
+                        if newItem.seen_at == nil {
+                            newItem.seen_at = date
+                        }
+                    case .read:
+                        if newItem.read_at == nil {
+                            newItem.read_at = date
+                        }
+                    case .interacted:
+                        if newItem.interacted_at == nil {
+                            newItem.interacted_at = date
+                        }
+                    case .archived:
+                        if newItem.archived_at == nil {
+                            newItem.archived_at = date
+                        }
```

Comment:
For the archive case we need to:

- Remove ALL entries when shouldHideArchived is true
- Reset ALL metadata counts when shouldHideArchived is true